taichi/cache/gfx/cache_manager.cpp

```diff
@@ -113,6 +249,9 @@ std::optional<CompiledKernelData> CacheManager::try_load_cached_kernel(
       auto compiled = aot_kernel_impl->params();
       // TODO: Support multiple SNodeTrees in AOT.
       compiled.num_snode_trees = compiled_structs_.size();
+      auto kmetadata = make_kernel_metadata(key, compiled);
+      offline_cache_metadata_.size += kmetadata.size;
```

Comment:
What is `offline_cache_metadata_` for? IIUC all the caches are read from `cached_module_` and the `offline_cache_metadata_` is never used in the program except comparing the version?

Reply:
> IIUC all the caches are read from cached_module_

Yes, you are right. The `offline_cache_metadata_: gfx::CacheManager::Metadata` is only used to save the offline cache meta data, which is useful for offline cache **but useless for AOT**.

The definition of `offline_cache::Metadata` is here:
ps. `gfx::CacheManager::Metadata` = `offline_cache::Metadata` now
https://github.com/taichi-dev/taichi/blob/1d25e1216db187336c08a7be7dd6e0a4a8715d6f/taichi/util/offline_cache.h#L48-L66

---

> the offline_cache_metadata_ is never used in the program except comparing the version of Taichi?

The offline_cache_metadata_ is used here:
https://github.com/taichi-dev/taichi/blob/1d25e1216db187336c08a7be7dd6e0a4a8715d6f/taichi/util/offline_cache.h#L124-L250

Reply:
The `offline_cache_metadata_` is mainly used by `CacheCleaner`

Reply:
Have you updated the AOT metadata after removing the spv files?

Reply:
Yes, here:
https://github.com/taichi-dev/taichi/blob/1d25e1216db187336c08a7be7dd6e0a4a8715d6f/taichi/cache/gfx/cache_manager.cpp#L60-L72

Reply:
Ok, I've found it in `save_metadata`